j2k/src/org/jetbrains/kotlin/j2k/ForConverter.kt

```diff
@@ -209,7 +209,7 @@ class ForConverter(
         return if (reversed)
             DownToExpression(startConverted, convertBound(bound, if (inclusiveComparison) 0 else +1))
         else
-            RangeExpression(startConverted, convertBound(bound, if (inclusiveComparison) 0 else -1))
+            UntilExpression(startConverted, convertBound(bound, if (inclusiveComparison) +1 else 0))
```

Comment:
I don't think this is correct. `until` should be used only in the case where the original code used `-1`. Cases where `0..9` was generated should remain unchanged.